### DISCUSSION TITLE
Module not found: Can't resolve 'fs' from getStaticProps

### DISCUSSION THREAD

**Discussion by reposman33** [vercel/next.js#24233] [Help]
Hello,
I started the LearnNext.js tutorial and am at "[Pre-rendering and Data Fetching](https://nextjs.org/learn/basics/data-fetching/implement-getstaticprops)". I call the getAllPosts function from index/getStaticProps that reads the 2 markdown files in the /posts directory but get an error instead:

```
./lib/posts.js:1:0
Module not found: Can't resolve 'fs'
> 1 | import fs from "fs";
  2 | import path from "path";
  3 | import matter from "gray-matter";
```

Seems like a server side dependency (fs) is being called client side... Afaics I followed the tutorial step by step so I do not know why this error occurs. The Github repos is here:

https://github.com/reposman33/nextjs-blog.git

Can anyone point out why I get this error when requesting a filelisting in getStaticProps?

**Comment by akellbl4** [comment]
I've fixed the problem. You have a typo in import: getS**p**ortedPostData instead of getSortedPostData
https://github.com/reposman33/nextjs-blog/pull/1

**Reply by reposman33** [reply]
oops! Thanks..

**Reply by akellbl4** [reply]
@reposman33 you are welcome! Please, mark the discussion as answered if the problem was solved ;)
